src/widget/wtrackmenu.cpp

```diff
@@ -897,6 +893,11 @@ void WTrackMenu::updateMenus() {
     if (featureIsEnabled(Feature::Properties)) {
         m_pPropertiesAct->setEnabled(singleTrackSelected);
     }
+
+    if (featureIsEnabled(Feature::FindOnWeb)) {
+        const auto pTrack = getFirstTrackPointer();
+        m_pFindOnMenu->setEnabled(!WFindOnWebMenu::hasEntriesForTrack(*pTrack));
```

Comment:
In C++ it is assumed that a reference can't be null. But this is only the case if the programmer takes care of it. 
In this case the pTrack can be null and the assumption is void. So please ass a check for null to fix that.

Reply:
So as far as I understood, before we set enable if the track has entries, we should check the pointer is null or not.  I think we should do that twice because we set enable two times first while creating the menu second while updating the menu right?

Reply:
Yes right. But I suggest to give the pointer to WFindOnWebMenu::hasEntriesForTrack and return false in that case. 
This way we have not much code duplication.

Reply:
@daschuer Passing a shared pointer is conceptually wrong if you do not plan to store it.

Reply:
No. It should be a const-ref that is also used by all other public methods. Passing a pointer is pointless and wrong. The whole menu would be obsolete without a valid `Track` object. Please don't confuse this!

Reply:
It is hard to guess which solution do you have in mind without more details. I don't understand why passing a raw pointer is pointless and wrong and doing the null check inside hasEntriesForTrack()

My suggestion was this: 

```suggestion
        m_pFindOnMenu->setEnabled(!WFindOnWebMenu::hasEntriesForTrack(pTrack.get()));
```

With a null check inside hasEntriesForTrack() 

Do you wish this solution? 

```suggestion
        const bool hasEntries = pTrack ? WFindOnWebMenu::hasEntriesForTrack(*pTrack) : false; 
        m_pFindOnMenu->setEnabled(!hasEntries);
```

or do you have something else in mind?

Reply:
WFindOnWebMenu must not be bothered with nullptr checks. This would be ridiculous.

Reply:
It does not need a pointer as an argument nor does it store one.

Reply:
Since we are talking here about a static function this argument is void. But anyhow, this is a matter of taste finally. You have not yet commented my alternatives. Is the second one OK to you?

Reply:
This is not a matter of taste and it doesn't matter if this is a static function or not. It's about scopes and responsibilities. The 2nd alternative is correct, because it keeps the nullptr check where it belongs. This also matches my initial proposal.

Reply:
`const auto enableMenu = pTrack && WFindOnWebMenu::hasEntriesForTrack(*pTrack);`

The name `hasEntries` is inappropriate in this context.

Reply:
You will hopefully notice that the implementation of `hasEntriesForTrack()` is also still wrong.

Reply:
What do you suggest? hasActions / hasSubmenus ...

Reply:
Ah, scratch that, you mean enableMenu instead of hasEntries for the local variable. Ok.

Reply:
> Yes, the whole PR suffers from unconditinal use of auto. This makes the code hard to review.

I disagree strongly. This isn't even an issue.

Reply:
You speak in riddles. Please give us more hints why the function is still wrong. 

> I disagree strongly. 

That is you personal preference and not part of our style guide. It even lists the following cases as bad:  
```const auto artist = track.getArtist();```
This is because in a review you cannot be certain what type auto is.